hoomd/hpmc/UpdaterRemoveDrift.h

```diff
@@ -43,10 +42,49 @@ template<class Shape> class RemoveDriftUpdater : public Updater
     public:
     //! Constructor
     RemoveDriftUpdater(std::shared_ptr<SystemDefinition> sysdef,
-                       std::shared_ptr<ExternalFieldLattice<Shape>> externalLattice,
-                       std::shared_ptr<IntegratorHPMCMono<Shape>> mc)
-        : Updater(sysdef), m_externalLattice(externalLattice), m_mc(mc)
+                       std::shared_ptr<IntegratorHPMCMono<Shape>> mc,
+                       pybind11::array_t<double> ref_positions)
+        : Updater(sysdef), m_mc(mc)
         {
+        setReferencePositions(ref_positions);
+        }
+
+    //! Set reference positions from a Nx3 numpy array
+    void setReferencePositions(const pybind11::array_t<double> ref_pos)
+        {
+        const size_t N = ref_pos.request().shape[0];
+        const size_t dim = ref_pos.request().shape[1];
+        if (N != this->m_pdata->getNGlobal() || dim != 3)
+            {
+            throw std::runtime_error("The array must be of shape (N_particles, 3).");
+            }
+        const double* rawdata = static_cast<double*>(ref_pos.request().ptr);
+        m_ref_positions.resize(m_pdata->getNGlobal());
+        for (size_t i = 0; i < N; i++)
+            {
+            const size_t array_index = i * 3;
+            this->m_ref_positions[i] = vec3<Scalar>(rawdata[array_index],
+                                                    rawdata[array_index + 1],
+                                                    rawdata[array_index + 2]);
+            }
+        }
+
+    //! Get reference positions as a Nx3 numpy array
+    pybind11::array_t<double> getReferencePositions() const
+        {
+        std::vector<size_t> dims(2);
+        dims[0] = this->m_ref_positions.size();
+        dims[1] = 3;
+        // the cast from vec3<Scalar>* to Scalar* is safe since vec3 is tightly packed without any
+        // padding. This also makes a copy so, modifications of this array do not effect the
+        // original reference positions.
+        const auto reference_array = pybind11::array_t<double>(
+            dims,
+            reinterpret_cast<const double*>(&(this->m_ref_positions[0])));
```

Comment:
```suggestion
            reinterpret_cast<const double*>(this->m_ref_positions.data()));
```
Cleaner access to the data pointer.